nixos/modules/config/networking.nix

```diff
@@ -11,6 +11,28 @@ let
 
   localhostMultiple = any (elem "localhost") (attrValues (removeAttrs cfg.hosts [ "127.0.0.1" "::1" ]));
 
+  localServicesFile =
+    pkgs.runCommand "etc-services" { preferLocalBuild = true; } (let
+      overrides = lib.pipe cfg.services [
+        (lib.attrsets.mapAttrsToList (name:
+          { port, protocols, aliases }:
+          (map (proto: "${name} ${toString port}/${proto} ${toString aliases}")
+            protocols)))
+        lib.lists.flatten
+        lib.strings.concatLines
+      ];
```

Comment:
How well does this scale? I am always a bit sceptical about mapAttrsToList. Could we maybe easily simplify this?

Reply:
It only iterates over the services entries that you have set in your configuration, which right now is zero.